js/dataset/Index.svelte

```diff
@@ -63,7 +63,7 @@
 		} else if (!samples) {
 			samples = [];
 		}
-		if (samples !== old_samples) {
+		if (JSON.stringify(samples) !== JSON.stringify(old_samples)) {
```

Comment:
Issue is that this was triggering infinitely many times because we're comparing two json objects